openedx/core/djangoapps/user_authn/views/tests/test_register.py

```diff
@@ -1573,6 +1574,27 @@ def test_country_overrides(self):
 
         self.assertContains(response, 'Kosovo')
 
+    def test_password_with_spaces(self):
+        """Test that spaces are stripped correctly from password while creating an account."""
+        unstripped_password = self.PASSWORD + '  '
+        with mock.patch(
+            'openedx.core.djangoapps.user_authn.views.register.authenticate_new_user',
+            wraps=authenticate_new_user
+        ) as mock_authenticate_new_user:
+            response = self.client.post(self.url, {
+                "email": self.EMAIL,
+                "name": self.NAME,
+                "username": self.USERNAME,
+                "password": unstripped_password,
+                "honor_code": "true",
+            })
+
+            mock_authenticate_new_user.assert_called_with(
```

Comment:
I think we should also test that user object is also returned.

Reply:
@adeelehsan I think that's unnecessary otherwise an exception has been raised and test would have failed.